### PR TITLE
WT-8070 Remove discrepancy between prefix_key and prefix_search

### PR DESCRIPTION
**Pull request by etienneptl**
This removes the discrepancy we had as we were using both `prefix_search` and `prefix_key` in our code.
It was decided in WT-8070 to keep `prefix_search`.

### CODE REVIEW

File: src/btree/bt_curnext.c [hunk 1]
```diff
@@ -348,7 +348,7 @@ __cursor_row_next(
     key = &cbt->iface.key;
     page = cbt->ref->page;
     session = CUR2S(cbt);
-    prefix_search = prefix != NULL && F_ISSET(&cbt->iface, WT_CURSTD_PREFIX_SEARCH);
```
Comment (jiechenbo) [inline src/btree/bt_curnext.c:351]: Nitpick: thinking to rename the `prefix_search` to `prefix_key_search`

**Review by jiechenbo — APPROVED** [review]
LGTM! Just one nitpick request

File: src/btree/bt_curnext.c (hunk 1 above)
Comment (etienneptl) [reply src/btree/bt_curnext.c:351]: I am fine with either, let's see what's Luke's preferences.

Comment (etienneptl) [reply src/btree/bt_curnext.c]: Please, review again as it was decided to keep `prefix_search` instead of `prefix_key` and the changes are not the same.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Review by jiechenbo — APPROVED** [review]
LGTM! This way looks much cleaner.

**Pull request merged**